src/React.Core/ReactComponent.cs

```diff
@@ -120,39 +120,42 @@ public virtual string RenderHtml(bool renderContainerOnly = false, bool renderSe
 				EnsureComponentExists();
 			}
 
-			try
+			var html = string.Empty;
+			if (!renderContainerOnly)
 			{
-				var html = string.Empty;
-				if (!renderContainerOnly)
+				try
 				{
 					var reactRenderCommand = renderServerOnly
 						? string.Format("ReactDOMServer.renderToStaticMarkup({0})", GetComponentInitialiser())
 						: string.Format("ReactDOMServer.renderToString({0})", GetComponentInitialiser());
 					html = _environment.Execute<string>(reactRenderCommand);
 				}
-
-				string attributes = string.Format("id=\"{0}\"", ContainerId);
-				if (!string.IsNullOrEmpty(ContainerClass))
+				catch (JsRuntimeException ex)
 				{
-					attributes += string.Format(" class=\"{0}\"", ContainerClass);
+					if (_configuration.ExceptionHandler == null) {
```

Comment:
I wonder if we should also allow setting an exception handler per component (eg. as an argument to this method). Hmm.